src/main/java/io/pravega/leaderelection/Leader.java

```diff
@@ -0,0 +1,28 @@
+package io.pravega.leaderelection;
+
+public class Leader {
+    private String leaderName;
+    private Leader() {
+        leaderName = null;
+    }
+
+    private static class LeaderSingleton {
+        private static final Leader INSTANCE = new Leader();
```

Comment:
This is introducing global state. This will break users in crazy an unexpected ways. Consider: It might be possible for an application to be running two different leader electors in the same process. Perhaps one is in a library. Or perhaps there are multiple unit tests for leader election, and we turn on parallelism to speed up our build. Then suddenly they interfere with one another and start setting incorrect state here.

What's more this is not thread safe, so it could be even more confusing.